compiler/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -615,12 +788,31 @@ fn resolve_trait_methods(
             // TODO
             let generics: Generics = vec![];
             let span: Span = name.span();
+            let default_impl_list: Vec<_> = unresolved_trait
+                .fns_with_default_impl
+                .functions
+                .iter()
+                .filter(|(_, _, q)| q.name() == name.0.contents)
+                .collect();
+            let default_impl = if !default_impl_list.is_empty() {
+                if default_impl_list.len() > 1 {
+                    // TODO(nickysn): Add check for method duplicates in the trait and emit proper error messages. This is planned in a future PR.
+                    panic!("Too many functions with the same name!");
+                }
```

Comment:
I'd think if there were duplicated methods on the trait, they should be verified before we start looking at the impl. So we shouldn't need the error at all here. Another reason not to include this is that if we did push an error here, we'd be pushing one for every impl rather than just once when checking the trait.

Reply:
I realize I didn't explain this clearly in the comment, but I plan to implement this check earlier and not here. I just postponed this for our next PR, because this PR is already too big. And we discovered we've missed this check too late in the development process of this PR. So, in a future PR (in fact, it is in my immediate TODO list after this PR gets merged) we will add this check at an earlier phase and we'll include tests and then this check might be removed completely or changed to unreachable (depending on the behaviour with the compile_failure tests that would be added - we want a single error message per trait and no compiler panic).